lgc/patch/PatchBufferOp.cpp

```diff
@@ -1645,6 +1657,53 @@ Instruction *PatchBufferOp::makeLoop(Value *const loopStart, Value *const loopEn
   return loopCounter;
 }
 
+// =====================================================================================================================
+// Provide well-defined behavior for accessing null descriptor.
+//
+// @param checkInst : The buffer descriptro user.
+void PatchBufferOp::processDescriptorUser(Value *checkInst) {
+  ExtractElementInst *extractInst = nullptr;
+  unsigned descIdx = 0;
+  Instruction *insertPos = nullptr;
+  if (isa<ExtractElementInst>(checkInst)) {
+    extractInst = dyn_cast<ExtractElementInst>(checkInst);
+    insertPos = extractInst;
+  } else {
+    // TODO: store and load instruction is to be handled
+    llvm_unreachable("Not implemented!");
+  }
+
+  if (extractInst) {
+    m_builder->SetInsertPoint(insertPos);
+    Value *bufferDesc = insertPos->getOperand(descIdx);
+
+    // Check whether is a null descriptor
+    Value *descIcmpVec = m_builder->CreateICmpEQ(bufferDesc, Constant::getNullValue(bufferDesc->getType()));
```

Comment:
Can we just check the first two dwords for 0 for a null descriptor?

Reply:
I'm not sure if it is sufficient.